src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -1,18 +1,505 @@
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationMirrorSet;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TreeUtils;
 
-public class DependencyInjectionAnnotatedTypeFactory extends ValueAnnotatedTypeFactory {
+public class DependencyInjectionAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
+
+  ClassValAnnotatedTypeFactory classValATF = getTypeFactoryOfSubchecker(ClassValChecker.class);
+
+  static HashMap<String, KnownBindingsValue> knownBindings = new HashMap<>();
+
+  /** The {@code com.google.inject.AbstractModule.bind(Class<Baz> clazz)} method */
+  private final List<ExecutableElement> bindMethods = new ArrayList<>(3);
+
+  /** The {@code com.google.inject.binder.LinkedBindingBuilder.to(Class<? extends Baz> method */
+  private final List<ExecutableElement> toMethods = new ArrayList<>(3);
+
+  /** The {@code com.google.inject.binder.LinkedBindingBuilder.toInstance() method */
+  private final List<ExecutableElement> toInstanceMethods = new ArrayList<>(1);
+
+  private final List<ExecutableElement> annotatedWithMethods = new ArrayList<>(2);
+
+  /** The ClassVal.value argument/element. */
+  public final ExecutableElement classValValueElement =
+      TreeUtils.getMethod(ClassVal.class, "value", 0, processingEnv);
+
+  /** The Bind.value argument/element. */
+  public final ExecutableElement bindValValueElement =
+      TreeUtils.getMethod(Bind.class, "value", 0, processingEnv);
+
+  /** The BindAnnotatedWith.value argument/element. */
+  public final ExecutableElement bawValValueElement =
+      TreeUtils.getMethod(BindAnnotatedWith.class, "value", 0, processingEnv);
+
+  /** The BindAnnotatedWith.annotatedWith argument/element. */
+  public final ExecutableElement bawAnnotatedWithValueElement =
+      TreeUtils.getMethod(BindAnnotatedWith.class, "annotatedWith", 0, processingEnv);
+
+  private void printKnownBindings() {
+    System.out.println("Known Bindings:");
+    DependencyInjectionAnnotatedTypeFactory.knownBindings.forEach(
+        (key, value) -> {
+          System.out.print(String.format("<Key: %s, Value: %s>\n", key, value));
+        });
+
+    System.out.println();
+  }
 
   public DependencyInjectionAnnotatedTypeFactory(BaseTypeChecker c) {
-    super(c);
-    super.postInit();
+    super(c, Bind.class, BindBottom.class);
+    ProcessingEnvironment processingEnv = this.getProcessingEnv();
+    this.bindMethods.add(
```

Comment:
Since we're talking about this, should we be using string literals for the `typeName` and `methodName` parameters of `TreeUtils.getMethod()`?

`this.bindMethods.add(TreeUtils.getMethod("com.google.inject.AbstractModule", "bind", processingEnv,"com.google.inject.Key<T>"));`

Reply:
Certainly for string literals that are repeated, like `"com.google.inject.AbstractModule"`, it would be better to put them in a variable. Overall, I'd say defining a few constants at the start of this class would make this a lot more readable.